src/main/java/net/snowflake/client/core/arrow/fullvectorconverters/ArrowFullVectorConverter.java

```diff
@@ -0,0 +1,125 @@
+package net.snowflake.client.core.arrow.fullvectorconverters;
+
+import java.util.Map;
+import net.snowflake.client.core.DataConversionContext;
+import net.snowflake.client.core.SFBaseSession;
+import net.snowflake.client.core.SFException;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.client.jdbc.SnowflakeType;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.types.Types;
+
+@SnowflakeJdbcInternalApi
+public interface ArrowFullVectorConverter {
```

Comment:
They are separate, I should probably write some comments about it. In short, the difference is that while `ArrowVectorConverters` allow for converting single values and are based on the initial type of the vector (so one input type, multiple output types), `ArrowVectorFullConverters` convert whole vectors at a time and are based on the type of the output vector (multiple input types, one output type). It is designed so that, in most cases, the combination of objects of these two classes allows for perfect translation between two types.

Reply:
A comment will be great